exchanges/kraken/kraken_wrapper.go

```diff
@@ -1078,7 +1392,51 @@ func (k *Kraken) GetHistoricCandlesExtended(pair currency.Pair, a asset.Item, st
 			Volume: candles[i].Volume,
 		})
 	}
-
 	ret.SortCandlesByTimestamp(false)
 	return ret, nil
 }
+
+// compatibleVars gets compatible variables for order vars
+func compatibleVars(side, direction, orderType, fillType string) (OrderVars, error) {
```

Comment:
I think you need to revisit this function. Side and direction are the same thing outside of casing. Also, if side or filltype are empty, you return an error and there are instances where you pass an empty string to this function! So you are forcing an error to occur!
eg
```
	vars, err := compatibleVars("", orderHistory.OrderEvents[o].Event.OrderCancelled.RecentOrder.Direction,
						orderHistory.OrderEvents[o].Event.OrderCancelled.RecentOrder.OrderType, "")
```
empty filltype ensures that it will always return an error

Reply:
On a second review after sleeping, I can see that its not throwing an error in the example. I think the function should still be rethought though. It seems like it would work better as 4 individual small functions that you call as necessary and not one big one where it has 7 different paths it can take